README rework (#166)
scaleway/serverless-scaleway-functions: README.md

- [Managing containers](docs/containers.md)
- [Configuring custom domains](docs/custom-domains.md)
- [Handling events (e.g. CRONs)](docs/events.md)
- [Security and secret management](docs/secrets.md)
- [Troubleshooting](docs/troubleshooting.md)

There are also language-specific notes for Serverless Functions:

- [Golang functions](docs/golang.md)
- [Javascript functions](docs/javascript.md)
- [PHP functions](docs/php.md)
- [Python functions](docs/python.md)
- [Rust functions](docs/rust.md)

## Configuration

Your functions are defined in the `serverless.yml` file.

Each `serverless.yml` file corresponds to one function _or_ container namespace.

The file format is described below:

```yml
# The name of your namespace
service: scaleway-python3

# Read environment variables from a .env file
useDotenv: false

# Use of this plugin. This must not be changed
plugins:
  - serverless-scaleway-functions

# Scaleway-specific configuration
provider:
  # Must not change
  name: scaleway

  # Function runtime
  # List: https://www.scaleway.com/en/docs/serverless/functions/reference-content/functions-lifecycle/#available-runtimes
  runtime: python310

>   # Global environment variables, used in every function

Reviewer: ```suggestion
  # Global environment variables, available for Container / Functions in this namespace
```

  env:
    MY_VAR: "some var"
    MY_OTHER_VAR: "some other var"

>   # Global secrets, used in every function

Reviewer: ```suggestion
  # Global secrets available for Container / Functions in this namespace
```

  secret:
    MY_SECRET: "some secret"
    MY_OTHER_SECRET: "some other secret"

  # Optional override of Scaleway credentials
  scwToken: <scw-token>
  scwProject: <scw-project-id>

  # Scaleway region for the deploy
  scwRegion: fr-par

# Include/exclude directories
package:
  patterns:
    - "!node_modules/**"
    - "!.gitignore"
    - "!.git/**"

# Define functions - cannot be used with custom.containers
functions:
  my-func:
    # Handler entrypoint
    handler: handler.py

>     # Minimum and maximum number of instances of the function

Reviewer: ```suggestion
    # Minimum and maximum number of instances
```

    minScale: 0
    maxScale: 10

>     # Memory limit for the function (in MiB)

Reviewer: ```suggestion
    # Memory limit (in MiB)
```

    # Limits: https://www.scaleway.com/en/docs/serverless/functions/reference-content/functions-limitations/
    memoryLimit: 1024

    # Maximum duration a request will wait to be served before it times out (in seconds)
    # Value in string format ex: "300s" (default: 300 seconds)
    timeout: 300s

    # Runtime for this function, allows overriding provider.runtime
    runtime: node20

>     # Whether to permit HTTP or HTTPS-only. Options: enabled, disabled, or redirected

Reviewer: ```suggestion
    # Whether to permit HTTP or HTTPS-only. Options: enabled, disabled (https only), or redirected (http -> https)
```

    httpOption: enabled

>     # Whether the container is public (no authentication), or private (uses token-based authentication)

Reviewer: ```suggestion
    # Controls the privacy: public (no authentication), or private (uses token-based authentication)
```

    privacy: public

    # Local environment variables, used only in this function
    env:
      LOCAL_VAR: "local var"

    # Local secrets, used only in this function
    secret:
      LOCAL_SECRET: "local secret"

    # Custom domains configured for the function
    # https://www.scaleway.com/en/docs/compute/functions/how-to/add-a-custom-domain-name-to-a-function/
    custom_domains:
      - my-func.some.domain.com

    # List of events to trigger the function
    events:
      - schedule:
          rate: "1 * * * *"
          # Data passed as input in the request
          input:
            key-a: "value-a"
            key-b: "value-b"

# Define containers, cannot be used with functions

> custom:

Reviewer: maybe separate both files ? It can be impressive to see a huge configuration file
